rust/crd/src/security.rs

```diff
@@ -492,4 +594,25 @@ impl<'a> KafkaTlsSecurity<'a> {
             format!("ssl.ca.location={cert_directory}/ca.crt"),
         ]
     }
+
+    fn kcat_client_sasl_ssl(
+        cert_directory: &str,
+        service_name: &str,
+        pod_fqdn: &String,
+    ) -> Vec<String> {
+        vec![
+            "-X".to_string(),
+            "security.protocol=SASL_SSL".to_string(),
+            "-X".to_string(),
+            format!("ssl.ca.location={cert_directory}/ca.crt"),
+            "-X".to_string(),
+            "sasl.kerberos.keytab=/stackable/kerberos/keytab".to_string(),
+            "-X".to_string(),
+            "sasl.mechanism=GSSAPI".to_string(),
+            "-X".to_string(),
+            format!("sasl.kerberos.service.name={service_name}"),
+            "-X".to_string(),
+            format!("sasl.kerberos.principal={service_name}/{pod_fqdn}@$KERBEROS_REALM"),
```

Comment:
I would prefer if we can get the principal to the following for consistency reasons with other product
`sasl.kerberos.principal=kafka/{kafka_name}@$KERBEROS_REALM`.
I'm happy to try that out as well, maybe a second pair of eyes can find a way to get that working.
But let's merge listener-op in first, it might change things

Reply:
Not sure if this should be our final solution, but the test now works when using broker listeners instead of the bootstrap: cb8301d.

Reply:
See c852a1b for changes with new listener for bootstrapper.